pkg/conditions/manager_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2025 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/fake"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name string
+		want Manager
+	}{{
+		name: "New returns a non-nil manager.",
+		want: &managerImpl{},
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := New(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("New() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_conditionSet_Mark(t *testing.T) {
```

Comment:
Nit: The prevalent pattern in our tests is to use names like `TestConditionSetMark` - no underscores in test names.

To be fair this is pure taste, but I'd rather stick to the established pattern. I've always found it a bit strange that Go style avoids underscores in names, except for test function names where they're fine. 🤔

Reply:
This is what goland generates and I think this style is to be able to see if the method you are testing is private or public, like if you had:

```
func (*Foo) Mark() {}
func (*Foo) mark() {}

// and maybe also

func (*foo) Mark() {}
func (*foo) mark() {}

```

To test these with the suggested style would be:

```
TestFooMark
TestFoomark
TestfooMark
Testfoomark
```

vs how goland will generate it:

```
Test_Foo_Mark
Test_Foo_mark
Test_foo_Mark
Test_foo_mark
```

In this style you know exactly what is public or private without conflicts.

Reply:
I'd prefer to stick to the prevalent pattern in the codebase. I see the rationale but AFAIK we haven't had a meaningful issue with conflicts/readability so far.

Generally - I'm open to changing the prevailing style if there's a strong reason to. This seems like a non-zero reason but I'm not convinced it's worth the churn to adopt.

Reply:
ok updated.